scopes/workspace/workspace/workspace.ts

```diff
@@ -1277,6 +1280,73 @@ export class Workspace implements ComponentFactory {
       await this.importObjects();
       this.logger.consoleSuccess();
     }
+    return this.installModules(options);
+  }
+
+  private async getComponentsWithDependencyPolicies() {
+    const allComponentIds = await this.listIds();
+    const componentConfigFiles: Record<string, ComponentConfigFile> = {};
+    const componentPoliciesById: Record<string, any> = {};
+    (
+      await Promise.all<ComponentConfigFile | undefined>(
+        allComponentIds.map((componentId) => this.componentConfigFile(componentId))
+      )
+    ).forEach((component) => {
+      if (component == null) return;
+      const depResolverAspect = component?.aspects.get(DependencyResolverAspect.id);
+      if (depResolverAspect == null || component == null) return;
+      const componentId = component.componentId.toString();
```

Comment:
I would recommend taking the compId from the allComponentIds and not from the config file. it's safer. (I saw few places where people by mistake change the comp id in the file). (sometime happen when someone create the component.json manually instead of by using the `eject-conf` cmd.
In order to do that, on your first promise all make sure to return not only the config file, but also the comp id you just visited.

Reply:
I can just read from allComponentIds directly using the index